### DISCUSSION TITLE
Program Caching and Verification

### DISCUSSION THREAD

**Discussion by nalum** [google/cel-go#1091] [Q&A] [answered]
Hey all, I'm wondering if there is a best practice around caching of the program and/or if there is a way to check the program against the source Cel Expression to see if it has changed and needs to be generated again?

**Comment by TristonianJones** [comment, marked as answer]
Hi @Nalum,

We haven't implemented support for this, but your best bet would be to create a checksum of the source expression to store alongside the compiled `cel.Program`. If you had a file monitor or polling task, you could validate the checksum of the proposed expression against the one which was cached. If they're different, then you'd want to recompile the program.

Cheers,

-Tristan

**Reply by nalum** [reply]
Thanks for the response @TristonianJones, appreciate it. Will keep this in mind when it comes time to implement :+1:
